blivet/arch.py

```diff
@@ -198,6 +199,29 @@ def is_cell():
     return False
 
 
+def is_t2mac():
+    """
+    :return: True if the hardware is an Intel-based Apple Mac with the T2 security chip, False otherwise.
+    :rtype boolean
+
+    """
+    APPLE_T2_PRODUCT_NAMES = {
+        "iMac20,2", "iMac20,1", "iMacPro1,1",
+        "MacPro7,1", "Macmini8,1", "MacBookAir9,1", "MacBookAir8,2",
+        "MacBookAir8,1", "MacBookPro16,4", "MacBookPro16,3", "MacBookPro16,2",
+        "MacBookPro16,1", "MacBookPro15,4", "MacBookPro15,3", "MacBookPro15,2",
+        "MacBookPro15,1",
```

Comment:
I didn't really like the idea of having a list of products here at first, but it's short enough and if Apple isn't really going to make more Intel Macs, I think it should be ok.

Reply:
Apple is not going to make more Intel macs, so this list should never expand. The other method would be to look for a PCI device that represents the T2, but it would have more complicated code, which I think would lead to more bugs.